iroha/docs/guides/develop-iroha-module.md

```diff
@@ -0,0 +1,148 @@
+# How-to develop a new Iroha Module
+
+When you need to add some functionality to the Iroha use this guide to develop a new Iroha Module.
+
+## Prerequisites
+
+* [Rust](https://www.rust-lang.org/tools/install)
+* Text Editor or IDE
+
+## Steps
+
+### Create new Rust module inside Iroha crate
+
+Inside `iroha/src/lib.rs` add declaration of your new module.
+For example for `bridge` module we add the following declaration:
+```rust
+#[cfg(feature = "bridge")]
+pub mod bridge;
+```
+
+So for you module `x` you will add `pub mod x;`. You should also place your new module under the 
+[Cargo feature](https://doc.rust-lang.org/cargo/reference/features.html) so other developers 
+will be able to turn it on and off when needed.
+
+Now create a separate file for your module, for `bridge` module it will be `iroha/src/bridge.rs`.
+So for your module `x` you will create a new file `iroha/src/x.rs`.
+
+### Add documentation
+
+Each module should provide description of own functionality via Rust Docs.
+
+In the start of the module file you should place docs block for the enclosing item.
+
+```rust
+//! Here you can see a good description of the module `x` and its functionality.
+```
+
+All public entites of your module should be documented as well. But first, let's create them.
+
+### Write your logic
+
+The development of a new Iroha Module has a goal - bring new functionality to the Iroha.
```

Comment:
Is is Module or module – I think it would be nice to have them consistent.
Again, why "the Iroha" – Iroha is one of a kind. Or if you mean an instance of Iroha, it might be better to use "a" and "instance"...